prometheus/internal/metric.go

```diff
@@ -46,6 +48,21 @@ func (s MetricSorter) Swap(i, j int) {
 	s[i], s[j] = s[j], s[i]
 }
 
+func (s MetricSorter) compare(si, sj string) bool {
+	// When the number of devices exceeds two digits, a sorting error occurs.
+	// dealing with sorting problems, especially CPU sorting errors
+	re := regexp.MustCompile(`\d+$`)
```

Comment:
Hm I think we can't use regex in such a hot path, we might need to be more smarter here if we ever do this ( let's talk on the issue) - or do it optionally.